BootloaderCommonPkg/Include/Library/CryptoLib.h

```diff
@@ -52,10 +62,31 @@ typedef UINT8 HASH_CTX[IPP_HASH_CTX_SIZE];   //IPP Hash context buffer
 // RSA verify required key parameter format
 //
 typedef struct {
-  UINT32                   Signature;
-  UINT8                    PubKeyData[0];
+  UINT32                   PubExp;
+  UINT8                    Modulus[0];
 } RSA_PUB_KEY;
 
+typedef struct {
+  UINT32                   Identifier;  //signature ('P', 'U', 'B', 'K')
+  UINT16                   KeySize;     //Length of Public Key
+  UINT8                    KeyType;     //RSA or ECC
+  UINT8                    Rsvd;
+  UINT8                    KeyData[0];  //Pubic key data(RSA_PUB_KEY) with KeySize bytes
+} PUB_KEY;
+
+typedef struct {
+  UINT32                   Identifier; //signature Identifier('S', 'I', 'G', 'N')
+  UINT16                   SigSize;    //Length of signature 2K and 3K in bytes
+  UINT8                    SigType;    //PKCSv1.5 or RSA-PSS or ECC
+  UINT8                    HashAlg;    //Hash Alg for signingh SHA256, SHA384
+  UINT8                    Signature[0];  //Signature length defined by SigSize bytes
+} SIGNATURE;
+
+
+#define RSA_SIGNATURE_SIZE         (RSA2048NUMBYTES + sizeof(SIGNATURE))
```

Comment:
These can be confusing as the name do not imply what kind of signature is this: 2K, 3K or anything else.
Why do we need them?

Reply:
Referring to RSA_SIGNATURE_SIZE, we would create RSA2048_SIGNATURE_SIZE  & RSA3072_SIGNATURE_SIZE   in subsequent patches.  This patch is kept as-is.

Reply:
We should calculate this length using SIGNATURE structure.

Reply:
Will add API/Functionality to calculate RSA_SIGNATURE_AND_KEY_SIZE n subsequent patch.